src/HeaderList.php

```diff
@@ -7,15 +7,23 @@ class HeaderList
     /** @var array */
     public $names;
 
+    /** @var bool */
+    public $headersSpecified = false;
+
     /**
      * @param array $names
      */
-    public function __construct(array $names)
+    public function __construct(array $names = null)
```

Comment:
Avoid creating "maybe types" like this where possible, it's easier to reason about how a variable operates if this can be avoided

Reply:
I've tried to avoid a Breaking Change, so this is tolerant. Unsure how to work around it.

Reply:
Moved to calling functions, resolved in https://github.com/99designs/http-signatures-php/pull/42/commits/5a5928c35fde1918a6f3320de3ed98f7845f65c4